transcom/mymove
# MAIN B-19789 frontend - office user ability to manage docs

## [B-19789](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19789)

INT PR: https://github.com/transcom/mymove/pull/13189

## Summary

This is the frontend work for giving the office users the ability to manage documents(orders, amended orders, and supporting documents). QAE/CSR can only view and will not be able to add or delete documents.
The UI design and the functionality to partially add orders was already implemented by another team and I've followed that design.

### How to test

1. As a customer -> create a move and finish submitting with any shipment.
2. As a SC -> find the move -> click `View and edit orders` button in move details page.
3. Use the Manage Orders and Manage Amended Orders button on the right sidebar to add/delete documents and see the changes reflected.
4. Go back to move details page and click the "Supporting Documents" tab.
5. Use the document component on the right sidebar to add/delete supporting documents.
6. Repeat for TOO/TIO.
7. QAE/CSR should not be able to add/delete documents.


## Screenshots
![image](https://github.com/transcom/mymove/assets/146856854/66e50763-3345-423b-8c3d-460b4f27be84)
![image](https://github.com/transcom/mymove/assets/146856854/4a51a260-219b-47c3-9fc0-ba9749fc62a4)
